docs/sources/panels-visualizations/query-transform-data/sql-expressions/index.md

```diff
@@ -134,22 +140,55 @@ The SQL expression workflow in Grafana is designed with the following behaviors:
 
 - **Non-tabular or incorrectly shaped data will not render in certain panels.** Visualizations such as graphs or gauges require properly structured data. Mismatched formats will result in rendering issues or missing data.
 
-For data to be used in SQL expressions, it must be in a **tabular format**, specifically the **FullLong format**. This means all relevant data is contained within a single table, with values such as metric labels stored as columns and individual cells. Because not all data sources return results in this format by default, Grafana will automatically convert compatible query results to FullLong format when they are referenced in a SQL expression.
-
 ## SQL conversion rules
 
-When a RefID is referenced within a SQL statement (e.g., `SELECT * FROM A`), the system invokes a distinct SQL conversion process.
+When you reference a RefID within a SQL statement (e.g., `SELECT * FROM A`), the system invokes a distinct SQL conversion process.
 
 The SQL conversion path:
 
-- The query result is treated as a single data frame, without labels, and is mapped directly to a tabular format.
+- The query result appears as a single data frame, without labels, and is mapped directly to a tabular format.
 - If the frame type is present and is either numeric, wide time series, or multi-frame time series (for example, labeled formats), Grafana automatically converts the data into a table structure.
 
-## Known limitations
+## Supported functions
 
-- Currently, only one SQL expression is supported per panel or alert.
-- Grafana supports certain data sources. Refer to [compatible data sources](#compatible-data-sources) for a current list.
-- Autocomplete is available, but column/field autocomplete is only available after enabling the `sqlExpressionsColumnAutoComplete` feature toggle, which is provided on an experimental basis.
+Grafana maintains a complete list of supported SQL keywords, operators, and functions in the SQL expressions query validator implementation.
+
+For the most up-to-date reference of all supported SQL functionality, refer to the `allowedNode` and `allowedFunction` definitions in the Grafana [codebase](https://github.com/grafana/grafana/blob/main/pkg/expr/sql/parser_allow.go).
+
+## Alerting and recording rules
+
+SQL expressions integrates alerting and recording rules, allowing you to define complex conditions and metrics using standard SQL queries. The system processes your query results and automatically creates alert instances or recorded metrics based on the returned data structure.
+
+For SQL Expressions to work properly with alerting and recording rules, your query must return:
+
+- One numeric column - **_required_**. This contains the value that triggers alerts or gets recorded.
+- One or more string columns - _optional_. These become **labels** for the alert instances or metrics. Column names and their row values become labels.
+
```

Comment:
There is a third requirement (for alerting and recording rules). That is that set strings column values must form a unique identifier for the row (open to better ways to say this!). Otherwise an error `duplicate_string_columns` is returned with a hint to use group by.

For example, if there are two rows like, the resulting label set for each row would be the same:
```
25,auth-service,us-east
12,auth-service,us-east
```
The whole request will fail if this happens. So best practice is to use `group by` statements in the SQL expression to ensure a unique combination of string values for the row.